airtable/emails.js

```diff
@@ -0,0 +1,229 @@
+// Global configuration -- insert global Nex sendgrid proxy credentials
+//==========================================================
+// This authenticates us to the Nex sendgrid proxy, a small Zeit function
+// that forwards our request to the Sendgrid API.
+// We need to keep this secret because anyone with this credential
+// can send email on our behalf
+const SENDGRID_PROXY_TOKEN= <insert credential here>
+
+//==========================================================
+// City-specific configuration: update to match the city
+//==========================================================
+
+// For each email type, point to the Sendgrid template ID
+// Find the Sendgrid template IDs in Sendgrid UI.
+const SENDGRID_TEMPLATES = {
+    "Matched: to Volunteer": <insert template id>
+    "Matched: to Recipient": <insert template id>
+}
+
+// todo: update paterson-specific
+const REPLY_TO = {
+    name: "Neighbor Express",
+    email: "neighborexpress@gmail.com"
+}
+
+// Customize what fields we pull in from a delivery for an email template.
+// If the city has adjusted field names, they need to be updated here too.
+
+async function getEmailInfoForDelivery(delivery) {
+    const vTable = base.getTable("Volunteers");
+    const allVolunteers = (await vTable.selectRecordsAsync()).records;
+    const volunteer = allVolunteers.find(v => v.id === delivery.getCellValue("Assigned volunteer")[0].id);
+    return {
+        delivery: {
+            name: delivery.getCellValue("Recipient Name"),
+            address: delivery.getCellValue("Address"),
+            details: delivery.getCellValue("Delivery Request Details"),
+            phone: delivery.getCellValue("Phone Number (D)"),
+            email: delivery.getCellValue("Email Optional (D)"),
+            instructions: delivery.getCellValue("Delivery Instructions"),
+            // we need to have a boolean value for each grocery store,
+            // in order to branch conditionally on the store in the email
+            grocery_store: {
+                shoprite: delivery.getCellValue("Grocery Store Name").name === "ShopRite, Little Falls",
+                golden_mango: delivery.getCellValue("Grocery Store Name").name === "Golden Mango, Paterson"
+            }
+        },
+        volunteer: {
+            name: volunteer.getCellValue("Name"),
+            phone: volunteer.getCellValue("Phone"),
+            email: volunteer.getCellValue("Email")
+        }
+    };
+}
+
+//==========================================================
+//==========================================================
+// Standard script begins here --
+// You shouldn't need to change anything below this line.
+//==========================================================
+
+// Define helper functions
+//==========================================================
+
+const matchedDeliveries = (await base.getTable("Deliveries").getView("Status: Matched").selectRecordsAsync()).records;
```

Comment:
I think github syntax highlighting breaks here because `await` isn't allowed at the top level in JS, but in the airtable env the whole thing is implicitly inside an `async` function? not sure